File: pascal_Triangle.java
// https://leetcode.com/problems/pascals-triangle/description/

class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> list = new ArrayList<>();
        for(int i = 1; i <= numRows; i++) {
            list.add(generateRow(i));
        }
        return list;
    }
    public static List<Integer> generateRow(int row) {
        List<Integer> li = new ArrayList<>();
        int ans = 1;
        li.add(ans);
        for(int i = 1; i < row; i++) {
            ans *= (row - i);
            ans /= i;
            li.add(ans);
        }
        return li;
    }
}

File: pascal_triangle.py
# https://leetcode.com/problems/pascals-triangle/description/

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        list = []
        for i in range(1, numRows + 1):
            temp = self.generateRows(i)
            list.append(temp)
        return list
    def generateRows(self, row: int) -> List[int]:
        li = []
        ans = 1
        li.append(ans)
        for i in range(1, row):
            ans *= (row - i)
            ans //= i
            li.append(ans)
        return li
